### ISSUE TITLE
Named swagger schema

### ISSUE THREAD

**Issue by andreafspeziale** [nestjs/terminus#2070]
### Is there an existing issue that is already proposing this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe it

Wouldn't be nice to have named schema in order to have a coherent look and feel like the others one?

Right now I see the following

<img width="1179" alt="Schermata 2022-11-05 alle 00 17 51" src="https://user-images.githubusercontent.com/7680940/200088252-d37fd8e2-c361-45f1-88ff-39565da2e081.png">

Vs a typical schema

<img width="1165" alt="Schermata 2022-11-05 alle 00 18 16" src="https://user-images.githubusercontent.com/7680940/200088279-d72d5838-b350-40d5-b9f2-c83dabd2719e.png">

Maybe you wont give it a name in order to avoid any kind of application schema name collision (?) Just wondering...

Thanks in advance!

### Describe the solution you'd like

Add `name` property here based on the `status`? I don't know if it will do the trick.

### Teachability, documentation, adoption, migration strategy

I believe it would be quite "impactless".

### What is the motivation / use case for changing the behavior?

Coherent look and feel of the generated swagger doc.

**Comment by BrunnerLivio** [comment]
@andreafspeziale Sounds good to me! Would you like to create a PR? :)

**Comment by andreafspeziale** [comment]
@BrunnerLivio `npm i/build` is broken on my side using `node v18.5.0`.

<img width="910" alt="Schermata 2022-11-05 alle 22 01 00" src="https://user-images.githubusercontent.com/7680940/200141201-07abba7b-6da8-4036-b1da-8d37b5bb1e64.png">

No problems by using `node v16.x.x`. Wouldn't be useful to specify engines in the `package.json`?

I've also start testing what I was proposing above but unfortunately it is not 100% what I would like to achieve.

Adding `title` will add the name to the schema

<img width="898" alt="Schermata 2022-11-06 alle 00 04 28" src="https://user-images.githubusercontent.com/7680940/200144770-c9cba214-8aad-4889-9829-8c7ec6cab56a.png">

but it will not possible to smoothly have it listed at the bottom

<img width="1220" alt="Schermata 2022-11-06 alle 00 05 50" src="https://user-images.githubusercontent.com/7680940/200144802-2df5bba2-eb8f-4065-8466-6420e52e6bb4.png">

I'm not sure how to achieve it but my feeling is that the only way would be to have the schemas as classes and import them into the `main.ts` of the consumer application as extra schemas.

```
const swaggerDocument = SwaggerModule.createDocument(app, swaggerConfig, {
  extraModels: [Exception],
});
```

Any suggestion to this brainstorming?

**Comment by BrunnerLivio** [comment]
@andreafspeziale 
> No problems by using node v16.x.x. Wouldn't be useful to specify engines in the package.json?
Yeah that'd be nice! Can you specify the version that is defined within the [`.nvmrc` file](https://github.com/nestjs/terminus/blob/master/.nvmrc)?



I haven't looked too deeply into it, but it [may be related to this issue](https://github.com/nestjs/swagger/issues/738).
Have you tried calling `ApiExtraModels` within [`health-check.decorator.ts`](https://github.com/nestjs/terminus/blob/master/lib/health-check/health-check.decorator.ts#L12) (similar how `ApiServiceUnavailableResponse` is called)

**Comment by andreafspeziale** [comment]
@BrunnerLivio Oh! I didn't notice the `nvm` file. Using that was smooth, so far so good!

I checked the issue u linked above and following your suggestion I did some tests but unfortunately I never saw the schemas rendering at the bottom of the swagger page.

Maybe I'm just misunderstanding your suggestion.

Could you try to be more specific? I will be happy to try it out on my side!

By the way I developed a similar use-case, as you can see from the previous comment screenshot I've an `Exception` schema in the schema list at the bottom of the swagger page.

I achieved it by exporting a class and a decorator from one of my monorepo packages named `common`.

As mentioned the package exports the class and a decorator to be used in the controllers:

```
export class Exception extends HttpException {
  constructor(response: ExceptionResponse, status: number) {
    super(response, status);
  }

  @ApiProperty()
  code!: string;

  @ApiProperty()
  message!: string;

  @ApiPropertyOptional({
    oneOf: [
      {
        type: 'array',
        items: {
          type: 'string',
        },
      },
      {
        type: 'object',
        additionalProperties: { type: 'unknown' },
      },
    ],
  })
  details?: string[] | Record<string, unknown>;
}
```

```
export const ApiExceptionResponse = ({
  status,
  description,
  example,
  examples,
}: ApiExceptionResponseOptions) =>
  applyDecorators(
    ApiResponse({
      status,
      description,
      content: {
        'application/json': {
          schema: { $ref: getSchemaPath(Exception) },
          example,
          examples,
        },
      },
    }),
  );
```

```
export class CatsController {
  constructor(private readonly catsService: CatsService) {}

  @Get(':id')
  @ApiOperation({
    summary: 'Retrieves a cat by id',
    description: 'A Cat is a sample entity. If not found throws a NotFoundException.',
  })
  @ApiExceptionResponse({
    status: 400,
    example: httpExceptionExamples.ValidationException.value,
  })
  @ApiExceptionResponse({
    status: 404,
    example: httpExceptionExamples.NotFoundException.value,
  })
  async getConfiguration(@Param() { id }: CatParam): Promise<Cat> {
    const { id: _, ...res } = await this.catsService.getCat(id);
    return res;
  }
}
```

```
...
import { Exception } from 'common';
...
const swaggerConfig = new DocumentBuilder()
  .setTitle(name)
  .setDescription(`${name} RESTful API service.`)
  .setVersion(version)
  .build();

const swaggerDocument = SwaggerModule.createDocument(app, swaggerConfig, {
  extraModels: [Exception],
});

SwaggerModule.setup('swagger', app, swaggerDocument);
```

Let me know your thoughts!

P.s: I noticed that the package contains some files that I would not include

<img width="170" alt="Schermata 2022-11-09 alle 22 14 39" src="https://user-images.githubusercontent.com/7680940/200943698-00b69daa-6770-4fa9-893a-806ef7de7be1.png">

Do you want me to open an issue and update the `[.npmignore](https://github.com/nestjs/terminus/blob/master/.npmignore)` ignoring:
- .nvmrc
- .husky
- docker-compose.yml

Cheers and sorry for the late response I've been working on my first [OS nestjs module](https://github.com/andreafspeziale/nestjs-memcached)!
